### PR TITLE
delete method of  gitea

### PR DESCRIPTION
**Pull request by MaazMS**
updated  delete method of  gitea.

### CODE REVIEW

**Comment by stakater-user** [comment]
@MaazMS Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/workshop-operator/commit/8bcc7d997c34ba187710ccdc1fd82d90d6b9de1c/checks) has Failed!

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-42d35470`

File: controllers/workshop_controller.go [hunk 1]
```diff
@@ -242,3 +244,15 @@ func (r *WorkshopReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&workshopv1.Workshop{}).
 		Complete(r)
 }
+
+func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop) (ctrl.Result, error) {
+	log := r.Log.WithValues("workshop", req.NamespacedName)
+	log.Info("Deleting workshop" + workshop.ObjectMeta.Name)
+
+	giteaNamespaceName := "gitea"
```
Comment (rasheedamir) [inline controllers/workshop_controller.go]: @MaazMS we shouldn't send namespace name from here! makes no sense. Lets rather define static constants in gitea controller itself

we shouldn't be leaking stuff like this! it's bad approach

Comment (MaazMS) [reply controllers/workshop_controller.go]: @rasheedamir  i will create constant for giteaNamespace Name in gitea controller

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-43bb4806`

File: controllers/workshop_controller.go [hunk 2]
```diff
@@ -242,3 +244,14 @@ func (r *WorkshopReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&workshopv1.Workshop{}).
 		Complete(r)
 }
+
+func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop) (ctrl.Result, error) {
+	log := r.Log.WithValues("workshop", req.NamespacedName)
+	log.Info("Deleting workshop" + workshop.ObjectMeta.Name)
+
+	if result, err := r.deleteGitea(workshop, giteaNamespaceName); util.IsRequeued(result, err) {
```
Comment (rasheedamir) [inline controllers/workshop_controller.go]: remove `giteaNamespaceName`

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-c7635f0c`

File: controllers/gitea_reconciler.go [hunk 3]
```diff
@@ -22,14 +26,26 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	GITEANAMESPACE       = "gitea"
+	GITEAOPERATORNAME    = "gitea-operator"
+	GITEASERVICENAME     = "gitea-service"
+	GITEACLUSTERROLENAME = "ClusterRole"
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: @MaazMS what about the role binding name?

File: controllers/gitea_reconciler.go [hunk 4]
```diff
@@ -100,21 +116,21 @@ func (r *WorkshopReconciler) addGitea(workshop *workshopv1.Workshop, users int,
...
 	// Wait for server to be running
-	if !kubernetes.GetK8Client().GetDeploymentStatus("gitea-server", giteaNamespace.Name) {
+	if !kubernetes.GetK8Client().GetDeploymentStatus(GITEASERVICENAME, giteaNamespace.Name) {
 		return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 1}, nil
 	}
 
 	// Extract app route suffix from openshift-console
 	giteaRouteFound := &routev1.Route{}
-	if err := r.Get(context.TODO(), types.NamespacedName{Name: "gitea-server", Namespace: giteaNamespace.Name}, giteaRouteFound); err != nil {
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: GITEASERVICENAME, Namespace: giteaNamespace.Name}, giteaRouteFound); err != nil {
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go:133]: @MaazMS this is `gitea-server` and not `gitea-service `

File: controllers/gitea_reconciler.go [hunk 5]
```diff
@@ -181,9 +197,8 @@ func createGitUser(workshop *workshopv1.Workshop, username string, giteaURL stri
 	return reconcile.Result{}, nil
 }
 
-/**
 // Delete Gitea
-func (r *WorkshopReconciler) deleteGitea(workshop *workshopv1.Workshop, users int, giteaNamespaceName string) (reconcile.Result, error) {
+func (r *WorkshopReconciler) deleteGitea(workshop *workshopv1.Workshop, giteaNamespaceName string) (reconcile.Result, error) {
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: @MaazMS  remove namespace name from the method! we don't need to pass it in method name

File: controllers/workshop_controller.go [hunk 6]
```diff
@@ -242,3 +244,14 @@ func (r *WorkshopReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&workshopv1.Workshop{}).
 		Complete(r)
 }
+
+func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop) (ctrl.Result, error) {
+	log := r.Log.WithValues("workshop", req.NamespacedName)
+	log.Info("Deleting workshop" + workshop.ObjectMeta.Name)
+
+	if result, err := r.deleteGitea(workshop, GITEANAMESPACE); util.IsRequeued(result, err) {
```
Comment (rasheedamir) [inline controllers/workshop_controller.go]: you don't need to pass the namespace name @MaazMS

File: controllers/gitea_reconciler.go [hunk 7]
```diff
@@ -22,14 +26,26 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	GITEANAMESPACE       = "gitea"
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: GITEANAMESPACENAME

File: controllers/gitea_reconciler.go [hunk 8]
```diff
@@ -100,21 +116,21 @@ func (r *WorkshopReconciler) addGitea(workshop *workshopv1.Workshop, users int,
...
+	giteaCustomResource := gitea.NewCustomResource(workshop, r.Scheme, GITEASERVICENAME, giteaNamespace.Name, labels)
 	if err := r.Create(context.TODO(), giteaCustomResource); err != nil && !errors.IsAlreadyExists(err) {
 		return reconcile.Result{}, err
 	} else if err == nil {
 		log.Infof("Created %s Custom Resource", giteaCustomResource.Name)
 	}
 
 	// Wait for server to be running
-	if !kubernetes.GetK8Client().GetDeploymentStatus("gitea-server", giteaNamespace.Name) {
+	if !kubernetes.GetK8Client().GetDeploymentStatus(GITEASERVICENAME, giteaNamespace.Name) {
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: it should be `GITEADEPLOYMENTNAME`

File: controllers/gitea_reconciler.go [hunk 9]
```diff
@@ -100,21 +116,21 @@ func (r *WorkshopReconciler) addGitea(workshop *workshopv1.Workshop, users int,
 	}
 
 	// Create Custom Resource
-	giteaCustomResource := gitea.NewCustomResource(workshop, r.Scheme, "gitea-server", giteaNamespace.Name, labels)
+	giteaCustomResource := gitea.NewCustomResource(workshop, r.Scheme, GITEASERVICENAME, giteaNamespace.Name, labels)
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: it should be `GITEADEPLOYMENTNAME`

File: controllers/gitea_reconciler.go (hunk 4 above)
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: it should be `GITEADEPLOYMENTNAME`

File: controllers/gitea_reconciler.go [hunk 10]
```diff
@@ -100,21 +116,21 @@ func (r *WorkshopReconciler) addGitea(workshop *workshopv1.Workshop, users int,
...
-	if !kubernetes.GetK8Client().GetDeploymentStatus("gitea-server", giteaNamespace.Name) {
+	if !kubernetes.GetK8Client().GetDeploymentStatus(GITEASERVICENAME, giteaNamespace.Name) {
 		return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 1}, nil
 	}
 
 	// Extract app route suffix from openshift-console
 	giteaRouteFound := &routev1.Route{}
-	if err := r.Get(context.TODO(), types.NamespacedName{Name: "gitea-server", Namespace: giteaNamespace.Name}, giteaRouteFound); err != nil {
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: GITEASERVICENAME, Namespace: giteaNamespace.Name}, giteaRouteFound); err != nil {
 		log.Errorf("Failed to find %s route", "gitea-server")
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go:139]: update here as well to it should be `GITEADEPLOYMENTNAME`

File: controllers/gitea_reconciler.go [hunk 11]
```diff
@@ -59,38 +75,38 @@ func (r *WorkshopReconciler) addGitea(workshop *workshopv1.Workshop, users int,
...
+	giteaClusterRole := kubernetes.NewClusterRole(workshop, r.Scheme, GITEAOPERATORNAME, giteaNamespace.Name, labels, kubernetes.GiteaRules())
 	if err := r.Create(context.TODO(), giteaClusterRole); err != nil && !errors.IsAlreadyExists(err) {
 		return reconcile.Result{}, err
 	} else if err == nil {
 		log.Infof("Created %s Cluster Role", giteaClusterRole.Name)
 	}
 
 	// Create Cluster Role Binding
-	giteaClusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, "gitea-operator", giteaNamespace.Name, labels, "gitea-operator", "gitea-operator", "ClusterRole")
+	giteaClusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, GITEAOPERATORNAME, giteaNamespace.Name, labels, GITEAOPERATORNAME, GITEAOPERATORNAME, GITEACLUSTERROLENAME)
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: @MaazMS it's not updated properly; the last one is kind

File: controllers/workshop_controller.go (hunk 2 above)
Comment (MaazMS) [reply controllers/workshop_controller.go]: i have removed giteaNamespaceName from deleteGitea

File: controllers/gitea_reconciler.go (hunk 3 above)
Comment (MaazMS) [reply controllers/gitea_reconciler.go]: This is parameter of role binding `giteaClusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, "gitea-operator", giteaNamespace.Name, labels, "gitea-operator", "gitea-operator", "ClusterRole")`  
There for i have create const name as 
```bash
GITEAOPERATORNAME    = "gitea-operator"
GITEASERVICENAME     = "gitea-service"
GITEACLUSTERROLENAME = "ClusterRole"
```

File: controllers/gitea_reconciler.go (hunk 5 above)
Comment (MaazMS) [reply controllers/gitea_reconciler.go]: yes

File: controllers/gitea_reconciler.go (hunk 7 above)
Comment (MaazMS) [reply controllers/gitea_reconciler.go]: yes , i do it.

**Comment by stakater-user** [comment]
@MaazMS Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-6e190609`

File: controllers/gitea_reconciler.go [hunk 12]
```diff
@@ -22,14 +26,31 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	GITEANAMESPACENAME    = "gitea"
+	GITEADEPLOYMENTNAME      = "gitea-service"
+	GITEAANSIBLEDEPLOYMENTNAME   = "gitea-operator"
+	CLUSTERROLENAME  = "ClusterRole"
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: @MaazMS what is this? it's kind and not name

File: controllers/gitea_reconciler.go [hunk 13]
```diff
@@ -51,46 +72,46 @@ func (r *WorkshopReconciler) addGitea(workshop *workshopv1.Workshop, users int,
...
-	giteaClusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, "gitea-operator", giteaNamespace.Name, labels, "gitea-operator", "gitea-operator", "ClusterRole")
+	giteaClusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, GITEAROLEBINDINGNAME, giteaNamespace.Name, labels, GITEAROLEBINDINGNAME, GITEAROLEBINDINGNAME, CLUSTERROLENAME)
 	if err := r.Create(context.TODO(), giteaClusterRoleBinding); err != nil && !errors.IsAlreadyExists(err) {
 		return reconcile.Result{}, err
 	} else if err == nil {
 		log.Infof("Created %s Cluster Role Binding", giteaClusterRoleBinding.Name)
 	}
 
-	giteaOperator := kubernetes.NewAnsibleOperatorDeployment(workshop, r.Scheme, "gitea-operator", giteaNamespace.Name, labels, imageName+":"+imageTag, "gitea-operator")
+	giteaOperator := kubernetes.NewAnsibleOperatorDeployment(workshop, r.Scheme, GITEAANSIBLEDEPLOYMENTNAME, giteaNamespace.Name, labels, imageName+":"+imageTag, GITEAANSIBLEDEPLOYMENTNAME)
```
Comment (rasheedamir) [inline controllers/gitea_reconciler.go]: @MaazMS wrong here!

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-ccaebfbb`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-f4d199f0`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-5dccb33c`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull stakater/workshop-operator:SNAPSHOT-PR-24-510caad8`

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**